spec/03-plugins/38-ai-proxy/02-openai_integration_spec.lua

```diff
@@ -1,10 +1,63 @@
 local helpers = require "spec.helpers"
 local cjson = require "cjson"
 local pl_file = require "pl.file"
+local pl_stringx    = require "pl.stringx"
 
 local PLUGIN_NAME = "ai-proxy"
 local MOCK_PORT = helpers.get_available_port()
 
+
+local FILE_LOG_PATH_STATS_ONLY = os.tmpname()
+local FILE_LOG_PATH_NO_LOGS = os.tmpname()
+local FILE_LOG_PATH_WITH_PAYLOADS = os.tmpname()
+
+
+local function wait_for_json_log_entry(FILE_LOG_PATH)
+  local json
+
+  assert
+    .with_timeout(10)
+    .ignore_exceptions(true)
+    .eventually(function()
+      local data = assert(pl_file.read(FILE_LOG_PATH))
+
+      data = pl_stringx.strip(data)
+      assert(#data > 0, "log file is empty")
+
+      data = data:match("%b{}")
+      assert(data, "log file does not contain JSON")
+
+      json = cjson.decode(data)
+    end)
+    .has_no_error("log file contains a valid JSON entry")
+
+  return json
+end
+
+local _EXPECTED_CHAT_STATS = {
+  openai = {
```

Comment:
I think we want to cover all supported providers and not only openai, right?

Reply:
Same that below  "The response is static" and transformed by the ai plugins for all providers.